azure-storage-common/src/Common/Internal/Validate.php

```diff
@@ -248,6 +248,40 @@ public static function isInstanceOf($objectInstance, $classInstance, $name)
         }
     }
 
+    /**
+     * Creates an anonymous function that checks if the given hostname is valid or not.
+     *
+     * @return callable
+     */
+    public static function getIsValidHostname()
+    {
+        return function ($hostname) {
+            return Validate::isValidHostname($hostname);
+        };
+    }
+
+    /**
+     * Throws an exception if the string is not of a valid hostname.
+     *
+     * @param string $hostname String to check.
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return boolean
+     */
+    public static function isValidHostname($hostname)
+    {
+        $isValid = filter_var($hostname, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME);
```

Comment:
Actually, `FILTER_VALIDATE_DOMAIN` is undefined in some PHP versions. Please make sure this code is compatible for 5.6 and all version > 7.0.

Reply:
Thank your for the review. I pushed a version which falls back to a regular expression on PHP <7.0. If you'd like a more uniform or stricter check, please let me know.